src/transformers/models/llama/modeling_llama.py

```diff
@@ -750,6 +750,7 @@ class KwargsForCausalLM(FlashAttentionKwargs, LossKwargs): ...
 class LlamaForCausalLM(LlamaPreTrainedModel, GenerationMixin):
     _tied_weights_keys = ["lm_head.weight"]
     _tp_plan = {"lm_head": "colwise_rep"}
+    _pp_plan = {"lm_head": {"input_keys": ["input"], "output_keys": ["logits"]}}
```

Comment:
Here does input account for input ids and input embeds?

Reply:
Good point for the `embed_tokens`!

Reply:
The LM head will only ever get hidden states from the end of a decoder.

If `BaseModel.forward` recieves `inputs_embeds` then the modeling code will choose to skip the `embed_tokens` layer. I'm not sure if this needs to be represented here, and if yes how.